### PR TITLE
Fix errors when a subreddit has <5 posts.

### LINKED ISSUE #701: IndexError: list index out of range
Sentry Issue: [SIR-LANCEBOT-4W](https://sentry.io/organizations/python-discord/issues/2356593012/?referrer=github_integration)

```
IndexError: list index out of range
  File "discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "bot/exts/evergreen/reddit.py", line 57, in get_reddit
    if posts[1]["data"]["over_18"] is True:

Unhandled command error: list index out of range
```

### PR DESCRIPTION
**Pull request by ChrisLovering**
## Relevant Issues

Closes #701

## Description

Changed a list slice from `[1]` to `[0]`.
Added a min check for length of the respond, before shuffling. 

## Reasoning

If a subreddit has <2 posts, the posts[1] check would fail with an IndexError.
If the subreddit had less than 5 posts, then the k=5 check would also error. These changes harden the command for these edge cases.

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] If dependencies have been added or updated, run `pipenv lock`?
- [x] **Lint your code** (`pipenv run lint`)?
- [x] Set the PR to **allow edits from contributors**?

### CODE REVIEW

**Review by ToxicKidz — APPROVED** [review]
:shipit:

**Review by Kronifer — APPROVED** [review]
:shipit:

File: bot/exts/evergreen/reddit.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ async def get_reddit(self, ctx: commands.Context, subreddit: str = 'python', sor
         if not posts:
             return await ctx.send('No posts available!')
 
-        if posts[1]["data"]["over_18"] is True:
+        if posts[0]["data"]["over_18"] is True:
             return await ctx.send(
                 "You cannot access this Subreddit as it is ment for those who "
```
Comment (Xithrius) [inline bot/exts/evergreen/reddit.py:59]:
```suggestion
                "You cannot access this Subreddit as it is meant for those who "
```

**Review by Xithrius — CHANGES_REQUESTED** [review]

**Review by Xithrius — APPROVED** [review]
It works well.

**Pull request merged**